### ISSUE TITLE
[br] Optional holiday - Dia do servidor público

### ISSUE THREAD

**Issue by rulrok** [vacanza/python-holidays#1451]
https://github.com/vacanza/python-holidays/blob/110e3cc6a7856f1064b4d27b08a96418285349b5/holidays/countries/brazil.py#L130

On brazilian holidays, some are considered 'optional' (feriado facultativo).
When verifying for a holiday, I'd like to get the fixed holidays, or filter out these 'optional' ones.

That is because, as the name suggests, its observance will depend on employer, region, government decision, etc.
https://blog-grancursosonline-com-br.translate.goog/feriados-pontos-facultativos/?_x_tr_sl=pt&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp

How could I handle a case like that?

**Comment by KJhellico** [comment]
@rulrok, thank you for your remark about Brazilian optional holidays.

As Python Holidays now support holidays categories, in PR #1452 we have implemented correct categorization of Brazil holidays. So now you can distinguish fixed and optional holidays.

Code example:
```python
>>> from holidays.countries.brazil import Brazil
>>> from holidays.constants import OPTIONAL, PUBLIC
>>> br = Brazil(years=2023)  # equivalent to: br = Brazil(categories=(PUBLIC,), years=2023)
>>> br_full = Brazil(categories=(OPTIONAL, PUBLIC), years=2023)
>>> "2023-10-28" in br
False
>>> "2023-10-28" in br_full
True
```
These changes are currently available in beta version, and version 0.33 will be released next week. I hope this will be useful.

**Comment by rulrok** [comment]
Nice. Thank you :)
